ansible/roles/ocp-workload-edge-deployments/tasks/workload.yml

```diff
@@ -0,0 +1,133 @@
+---
+# Implement your Workload deployment tasks here
+
+- name: Setting up workload for user
+  debug:
+    msg: "Setting up workload for user ocp_username = {{ ocp_username }}"
+
+- name: Create project for IoT Development
+  shell: "oc new-project iot-development"
+
+- name: Create resources for IoT Development
+  shell: "oc process -f /tmp/{{guid}}/project-iot-development.yaml | oc create -f -"
+
+- name: Create project for IoT Testing
+  shell: "oc new-project iot-testing"
+
+- name: Create resources for IoT Testing
+  shell: "oc process -f /tmp/{{guid}}/project-iot-testing.yaml | oc create -f -"
+
+- name: Create project for IoT Hub
+  shell: "oc new-project iot-hub"
+
+- name: Create resources for IoT Hub
+  shell: "oc process -f /tmp/{{guid}}/project-iot-hub.yaml | oc create -f -"
+
+- name: Create policy for letting Jenkins ServiceAccount Access to iot-testing Project
+  shell: "oc policy add-role-to-user edit system:serviceaccount:iot-development:jenkins -n iot-testing"
+
+- name: Generate a brand new SSH key for connecting to the remote machine
+  shell: "cd /tmp/{{guid}}/deploy-containers-apb; ssh-keygen -N '' -f ./id_rsa"
```

Comment:
Do you really need to create a new ssh key ?
the bastion should be able to access all the instances, as root and using the internaldns name.

Also, you're using `id_rsa.pub` later, so you need

```shell
ssh-keygen -f ./id_rsa -y > id_rsa.pub
```

Reply:
I prefer to create and use a brand new key because the default key will allow Demo users to connect to any OCP nodes. Generating a brand new key, adding it ONLY to the authorized_keys of the bastion will ensure that user can use the key only for connecting to the bastion.
Is that ok for you?

Reply:
Sure